src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeRepository.java

```diff
@@ -23,7 +23,7 @@
     @RestResource(exported = false)
     SubmissionEnvelope findByUuid(Uuid uuid);
 
-    @RestResource(rel = "findByUuid")
+    @RestResource(rel = "findByUuid", path = "findByUuid")
     SubmissionEnvelope findByUuidUuid(@Param("uuid") UUID uuid);
```

Comment:
Why does `Uuid` appear twice in the method name?

Reply:
Probably because of the non-exported `findByUuid` method?

Reply:
A bit of a late response here but i can explain, this is due to the automated spring mongo methods interacting with our `Uuid` class.

`SubmissionEnvelope` extends [`AbstractEntity`](https://github.com/ebi-ait/ingest-core/blob/9c47f53a5264f0a4537e9dbd3e320e6beede742d/src/main/java/org/humancellatlas/ingest/core/AbstractEntity.java#L41) which contains the `uuid` property of type `Uuid` from `org.humancellatlas.ingest.core.Uuid`.
The [`Uuid`](https://github.com/ebi-ait/ingest-core/blob/9c47f53a5264f0a4537e9dbd3e320e6beede742d/src/main/java/org/humancellatlas/ingest/core/Uuid.java#L16) class contains a single property `uuid` of type `UUID` from `java.util.UUID`.

So using a snippet of this example, you can see that the uuid property of the submissionEnvelope is an object containing a uuid property.
[`GET /submissionEnvelopes/6080a6a1dd9aab1232d308b0`](https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/6080a6a1dd9aab1232d308b0)
```
{
  "submissionDate" : "2021-04-21T22:26:41.995Z",
  "updateDate" : "2021-04-22T00:07:18.005Z",
  "user" : "601c2af3ac1792031eee4261",
  "lastModifiedUser" : "anonymousUser",
  "type" : "Submission",
  "uuid" : {
    "uuid" : "9262e9bf-5ebd-44ee-b80f-15cf107b32a5"
  },
  "events" : [ ]
}
```
So when using mongo spring repositories, the name of the method reflects where the data is being pulled from the source document.
The non-exported `findByUuid ` method would be used with an object containing a uuid property:
```
uuid_object = {"uuid": "9262e9bf-5ebd-44ee-b80f-15cf107b32a5"}
envelope = submissionEnvelopeRepository.findByUuid(uuid_object);
```
whereas the second method just takes the UUID, which is more usable in our rest api, Hal browser, called from ingest UI.
```
uuid = "9262e9bf-5ebd-44ee-b80f-15cf107b32a5"
envelope = submissionEnvelopeRepository.findByUuidUuid(uuid);
```

I common language whenever anyone on the team talks about the uuid they mean this "9262e9bf-5ebd-44ee-b80f-15cf107b32a5" but actually in our code that is the Uuid of the Uuid. 

Why do we have a custom Uuid class? I don't know.